exchange/exchange_test.go

```diff
@@ -2547,6 +2547,108 @@ func TestCategoryMappingTwoBiddersOneBidEachNoCategorySamePrice(t *testing.T) {
 
 }
 
+func TestCategoryMappingTwoBiddersManyBidsEachNoCategorySamePrice(t *testing.T) {
```

Comment:
The large amount of test bids and iteration of 100 to workaround the random aspect makes it very hard to understand the test setup. Would this be a good time to isolate the random library and substitute it for just this test (for now) for a more targeted setup?

Reply:
Yes, refactored.